### PR TITLE
Add support for different languages

### PR DESCRIPTION
**Pull request by bazelajoanna**
Add i18next and translations

### CODE REVIEW

File: public/locales/pl/translation.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+  "common": {
+    "logoAlt": "Logo aplikacji",
+    "scrollIconTitle": "Ikona przewijania strony do początku",
+    "downloadIcon": "Ikona pobierania",
```
Comment (franiakat) [inline public/locales/pl/translation.json]: Could we make it consistent, so either always use only Icon not adding "title" "label" "aria"? Or are those cases somehow different?

File: public/locales/pl/translation.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "moderatorPageLink": "Strefa moderatora",
+    "hamburgerAlt": "Otwórz menu",
+    "menuIconAlt": "Ikona menu"
+  },
+  "problemDetailedPage": {
+    "goBackToProblemsList": "Powrót do listy problemów",
+    "downloadProblem": "Pobierz treść",
+    "problemContent": "Treść problemu",
+    "openingGuidance": "Propozycja wprowadzenia do zajęć",
+    "ressources": "Pobierz materiały dodatkowe",
```
Comment (franiakat) [inline public/locales/pl/translation.json]:
```suggestion
    "resources": "Pobierz materiały dodatkowe",
```

Comment (franiakat) [reply public/locales/pl/translation.json]: BTW it's added here twice

File: public/locales/pl/translation.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    "problemExtension": "Rozszerzenie problemu",
+    "problemExtensionContent": "Treść rozszerzenia",
+    "answer": "Odpowiedź",
+    "resources": "Pobierz materiały dodatkowe",
+    "guidance": "Wskazówki",
+    "closingGuidance": "Propozycja podsumowania zajęć"
+  },
+  "problemsPage": {
+    "inputHeader": "Szukaj problemów matematycznych",
```
Comment (franiakat) [inline public/locales/pl/translation.json]: I'd name it more descriptively `searchInputHeader`?

File: public/locales/pl/translation.json (hunk 1 above)
Comment (bazelajoanna) [reply public/locales/pl/translation.json]: You're right, I've removed "title".

File: src/App.test.tsx [hunk 4]
```diff
@@ -7,49 +7,47 @@ import App from "./App";
...
+    const moderatorPageNavLink = screen.getByText("moderatorPageLink");
     userEvent.click(moderatorPageNavLink);
-    const logo = screen.getAllByAltText(/Logo aplikacji/i);
+    const logo = screen.getAllByAltText("logoAlt");
     userEvent.click(logo[0]);
 
-    expect(
-      screen.getByText("Szukaj problemów matematycznych")
-    ).toBeInTheDocument();
+    expect(screen.getByText("inputHeader")).toBeInTheDocument();
```
Comment (evulins) [inline src/App.test.tsx]: Here also we need to change `"inputHeader" `for `"searchInputHeader"`

File: src/App.tsx [hunk 5]
```diff
@@ -20,7 +20,7 @@ const App: FC = () => {
         <Route path="/moderator" element={<Moderator />} />
       </Routes>
       <Footer />
-    </>
+    </Suspense>
```
Comment (evulins) [inline src/App.tsx:23]: 👍

File: src/components/organisms/ProblemSearchSection/index.tsx [hunk 6]
```diff
@@ -21,7 +24,7 @@ const ProblemSearchSection: FC<ProblemSearchSectionProps> = ({
       padding="2.4rem 1.6rem"
       paddingDT="2.4rem"
     >
-      <S.H2>Szukaj problemów matematycznych</S.H2>
+      <S.H2>{t("inputHeader")}</S.H2>
```
Comment (evulins) [inline src/components/organisms/ProblemSearchSection/index.tsx]: Same here :) "inputHeader" ->  "searchInputHeader"

**Review by evulins — COMMENTED** [review]
Wow 🍸  This is a new level of our work! Great success :)

File: src/App.test.tsx (hunk 4 above)
Comment (bazelajoanna) [reply src/App.test.tsx]: Thanks!

**Review by evulins — APPROVED** [review]

**Review by franiakat — APPROVED** [review]
Looks legit! 👏 😎 ready to be
sqash

**Pull request merged**